docs/component-specs.md

```diff
@@ -1,12 +1,65 @@
 # Spec files for agent-managed components
 
-Spec files are YML files that describe the capabilities of a component and when / how to run it. They are used by Agent to convert a user-provided policy into a configured running process.
+Spec files are YML files that describe the capabilities of a component and when / how to run it. They are used by Agent to convert a user-provided [policy](agent-policy.md) into a configured running process.
 
-[This documentation should be expanded](https://github.com/elastic/elastic-agent/issues/2690)
+## Spec file layout
 
-## Preventions
+A spec file corresponds to a specific executable, which must be in the same directory and have the same name -- for example `filebeat.spec.yml` would correspond to the executable `filebeat`, or `filebeat.exe` on Windows. The configuration is broken into sections:
 
-Components may include a `runtime.preventions` section containing [EQL conditions](https://www.elastic.co/guide/en/elasticsearch/reference/current/eql-syntax.html#eql-syntax-conditions) which should prevent the use of that component if any are true. Each prevention should include a `condition` in EQL syntax and a `message` that will be displayed if the condition prevents the use of a component.
+```yml
+version: 2
+inputs:
+  - name: <input name 1>
+    ...
+  - name: <input name 2>
+    ...
+shippers:
+  - name: <shipper name 1>
+    ...
+```
+
+The `version` key must be present and must equal 2 (to distinguish from the older version 1 schema that is no longer supported).
+
+`inputs` is a list of input types this component can run, and `shippers` is a list of shipper types this component can run. Each configured input and shipper also has its own list of `outputs` that it supports, but the spec file only tracks the list of supported types, and the rest comes from the [Agent policy](agent-policy.md).
```

Comment:
what are different shipper types?can be this per input?

Reply:
> looking at it it looks nice, documentation 🪨 🪨 I love breaking down the code into smaller units to increase readability. before approving i'd like to see [Blakes comment](https://github.com/elastic/elastic-agent/pull/2728/files#r1213323278) resolved/answered

@michalpristas I'm pretty sure Craig answered it in https://github.com/elastic/elastic-agent/pull/2728#issuecomment-1572282594